### DISCUSSION TITLE
Unable to find from Branch - fatal: Not a valid object name HEAD^\n

### DISCUSSION THREAD

**Discussion by anustupgit** [scolladon/sfdx-git-delta#670] [General]
Hi @scolladon , my code is unable to find **from** branch and it errors out with  : 
{
  "error": "fatal: Not a valid object name HEAD^\n",
  "output": ".",
  "success": false,
  "warnings": []
}

Below is my yml file :

pr:
  autoCancel: "true"
  branches:
    include:

    - QA
  paths:
    exclude:
    - README.md
    - SFDQ.yml
stages:
- stage: TestSalesforceOrg
  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/QA'))
  jobs:
  - job: DeployTestOrg
    steps:
      - task: NodeTool@0
        inputs:
          versionSpec: '16.x'
          checkLatest: true
      - bash: 
          npm install sfdx-cli --global
        displayName: Install Salesforce CLI
      - bash:
            sfdx config:set restDeploy=false --global
        displayName: Use SOAP API for Deployment
      - bash:

            sfdx force:auth:jwt:grant --clientid ************************************

        displayName: Authorize Salesforce DeployTestOrg Org
      - bash: echo 'y' | sfdx plugins:install sfdx-git-delta
        displayName: Installing sfdx-git-delta
      - bash: sfdx sgd:source:delta --to "HEAD" --from HEAD^ --output .
        displayName: Checking for delta change
      - bash: cat package/package.xml
        displayName: Creating Delta Package
      - bash: cat destructiveChanges/destructiveChanges.xml
        displayName: Creating Delta Destructive Package
      - bash:
            sfdx force:source:convert --manifest=package/package.xml --outputdir=deltaChange
        displayName: Convert Delta Code
 
      - bash:
            sfdx force:source:deploy -p deltaChange -w 10 -u *************************************
        displayName: Deploy source code to DeployTestOrg Org

**Comment by scolladon** [comment]
Hi @anustupgit,

We have a [dedicated section](https://github.com/scolladon/sfdx-git-delta#cicd-specificity) about CI specificities to help deal with all those aspects.

Does this pipeline fetch all the history or only the last commit ?

**Comment by Mark-Kloudblaze** [comment]
I just faced the same issue with Azure Pipelines.
I use the following script: 
```
          - bash: git fetch origin YOUR-TARGET-BRANCH
            displayName: Fetching YOUR-TARGET-BRANCH branch
          - bash: sfdx sgd:source:delta --from HEAD~1 --output .
```
What helped is ensuring that I fetch the history as Azure by default uses [Shallow Fetch](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines)

so you have to ensure to overwrite shallow fetch at the job level:
```
    jobs:
      - job: deploy
        displayName: Deploy - YOUR TARGET BRANCH
        steps:
          - checkout: self
            persistCredentials: true
            fetchDepth: 0
```
